### PR TITLE
dev-java/gson: add 2.11.0

### PR DESCRIPTION
**Pull request by vaukai**
---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Review by fordfrog — APPROVED** [review]

File: dev-java/gson/gson-2.11.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+JAVA_SRC_DIR=(
+	"src/main/java"
+	"src/main/java-templates"
+)
+
+JAVA_TEST_EXCLUDES=(
+	# requires the test class to be obfuscated using proguard which we do not have atm
+	"com.google.gson.functional.EnumWithObfuscatedTest"
+	# testComGoogleGsonAnnotationsPackage(com.google.gson.regression.OSGiTest)
+	# junit.framework.AssertionFailedError: Cannot find com.google.gson OSGi bundle manifest
```
Comment (fordfrog) [inline dev-java/gson/gson-2.11.0.ebuild:54]: i guess this means we are not putting osgi bundle manifest in the `META-INF/MANIFEST.MF` file, right? if so, we could fix this.

File: dev-java/gson/gson-2.11.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	sed -e "s/\${project.version}/${PV}/g" \
+		-i src/main/java-templates/com/google/gson/internal/GsonBuildConfig.java \
+			|| die "Failed to set version"
+}
+
+src_compile() {
+	if ! use test; then
+		JAVA_PKG_WANT_SOURCE="1.8"
+		JAVA_PKG_WANT_TARGET="1.8"
+	fi
```
Comment (fordfrog) [inline dev-java/gson/gson-2.11.0.ebuild:77]: why do we force here source/target 1.8 when we never allow the package to run java lower than 17?

**Review by fordfrog — CHANGES_REQUESTED** [review]

File: dev-java/gson/gson-2.11.0.ebuild (hunk 1 above)
Comment (vaukai) [reply dev-java/gson/gson-2.11.0.ebuild:54]: maybe. i don't even know what parts (lines) belong to `osgi bundle manifest`. upstream seems to build it from a `bnd.bnd` file. replicating this in the ebuild would require much better understanding of how to use `bnd`. it's not getting clear for me by reading `mvn's` log.

File: dev-java/gson/gson-2.11.0.ebuild (hunk 2 above)
Comment (vaukai) [reply dev-java/gson/gson-2.11.0.ebuild:77]: guess we could drop that condition will check this.

**Pull request closed without merging**